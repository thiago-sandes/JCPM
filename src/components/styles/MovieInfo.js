import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  header: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingBottom: 40,
  },
  prod: {
    fontWeight: 'bold',
    fontSize: 25,
    marginBottom: 8,
  },
  lower: {
    alignItems: 'flex-start',
    marginLeft: 8,
  },
  text: {
    fontSize: 17,
    padding: 5,
  },
  shadow: {
    shadowColor: '#202020',
    shadowOffset: {width: 5, height: 5},
    shadowRadius: 5,
    shadowOpacity: 0.39,
    elevation: 2,
  },
  img: {
    width: 210,
    height: 290,
    borderRadius: 7,
  },
  titleText: {fontSize: 20, fontWeight: 'bold', padding: 10},
  releasedText: {fontSize: 17},
  boldText: {fontWeight: 'bold'},
});
