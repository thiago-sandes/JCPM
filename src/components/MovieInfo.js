import React from 'react';
import {View, Text, Image} from 'react-native';

// Styles
import styles from './styles/MovieInfo';

const MovieInfo = (props) => {
  const {details} = props;
  return (
    <>
      <View style={styles.header}>
        <View style={styles.shadow}>
          <Image source={{uri: `${details.Poster}`}} style={styles.img} />
        </View>
        <Text style={styles.titleText}>{details.Title}</Text>
        <Text style={styles.releasedText}>{details.Released}</Text>
      </View>
      <View style={styles.lower}>
        <Text style={styles.text}>
          <Text style={styles.boldText}>Gênero: </Text> {details.Genre}
        </Text>
        <Text style={styles.text}>
          <Text style={styles.boldText}>Duração: </Text>
          {details.Runtime}
        </Text>
        <Text style={styles.text}>
          <Text style={styles.boldText}>Sinopse: </Text>
          {details.Plot}
        </Text>
        <Text style={[styles.text, {fontWeight: 'bold'}]}>Avaliação:</Text>
        {details.Ratings[0] && (
          <Text style={styles.text}>
            {details.Ratings[0].Source}: {details.Ratings[0].Value}
          </Text>
        )}
        {details.Ratings[1] && (
          <Text style={styles.text}>
            {details.Ratings[1].Source}: {details.Ratings[1].Value}
          </Text>
        )}
        {details.Ratings[2] && (
          <Text style={styles.text}>
            {details.Ratings[2].Source}: {details.Ratings[2].Value}
          </Text>
        )}
      </View>
    </>
  );
};

export default MovieInfo;
