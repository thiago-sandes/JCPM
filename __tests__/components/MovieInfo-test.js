import React from 'react';
import renderer from 'react-test-renderer';
import MovieInfo from '../../src/components/MovieInfo';

test('MovieInfo snapshot', async () => {
  const detailsMock = {
    Title: 'Avenger',
    Released: '09 Apr 2006',
    Runtime: '92 min',
    Genre: 'Thriller',
    Plot:
      'A C.I.A. Agent steps in to stop a former Special Forces Operative on a for-hire mission that poses a global threat.',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMTMzMjMwMjcyNl5BMl5BanBnXkFtZTcwNTA1NDgzMQ@@._V1_SX300.jpg',
    Ratings: [
      {Source: 'Internet Movie Database', Value: '5.7/10'},
      {Source: 'Rotten Tomatoes', Value: '29%'},
    ],
  };

  const tree = renderer.create(<MovieInfo details={detailsMock} />).toJSON();

  expect(tree).toMatchSnapshot();
});
