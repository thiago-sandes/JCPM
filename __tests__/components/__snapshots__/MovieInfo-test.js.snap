// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingBottom": 40,
      }
    }
  >
    <View
      style={
        Object {
          "elevation": 2,
          "shadowColor": "#202020",
          "shadowOffset": Object {
            "height": 5,
            "width": 5,
          },
          "shadowOpacity": 0.39,
          "shadowRadius": 5,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://m.media-amazon.com/images/M/MV5BMTMzMjMwMjcyNl5BMl5BanBnXkFtZTcwNTA1NDgzMQ@@._V1_SX300.jpg",
          }
        }
        style={
          Object {
            "borderRadius": 7,
            "height": 290,
            "width": 210,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
          "padding": 10,
        }
      }
    >
      Avenger
    </Text>
    <Text
      style={
        Object {
          "fontSize": 17,
        }
      }
    >
      09 Apr 2006
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "marginLeft": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 17,
          "padding": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Gênero: 
      </Text>
       
      Thriller
    </Text>
    <Text
      style={
        Object {
          "fontSize": 17,
          "padding": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Duração: 
      </Text>
      92 min
    </Text>
    <Text
      style={
        Object {
          "fontSize": 17,
          "padding": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Sinopse: 
      </Text>
      A C.I.A. Agent steps in to stop a former Special Forces Operative on a for-hire mission that poses a global threat.
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontSize": 17,
            "padding": 5,
          },
          Object {
            "fontWeight": "bold",
          },
        ]
      }
    >
      Avaliação:
    </Text>
    <Text
      style={
        Object {
          "fontSize": 17,
          "padding": 5,
        }
      }
    >
      Internet Movie Database
      : 
      5.7/10
    </Text>
    <Text
      style={
        Object {
          "fontSize": 17,
          "padding": 5,
        }
      }
    >
      Rotten Tomatoes
      : 
      29%
    </Text>
  </View>,
]
`;
